master/pyext/src/validation/Report.py

```diff
@@ -8,301 +8,337 @@
-			except:
-				Template_Dict['parameters_volume']=utility.dict_to_JSlist(I_sas.get_parameters_vol_many_dep())
-			try:
-				Template_Dict['parameters_mw']=utility.dict_to_JSlist(I_sas.get_parameters_mw_many())
-			except:
-				Template_Dict['parameters_mw']=utility.dict_to_JSlist(I_sas.get_parameters_mw_many_dep())
-			Template_Dict['pddf_info']=utility.dict_to_JSlist(I_sas.get_pddf_info())
-			Template_Dict['number_of_fits']=I_sas.get_total_fits()
-			Template_Dict['chi_table']=utility.dict_to_JSlist(I_sas.get_chi_table())
-			Template_Dict['rg_table']=utility.dict_to_JSlist(I_sas.get_rg_table_many())
-			Template_Dict['sasdb_code_fits']=I_sas.get_sasdb_code_fits()
-			Template_Dict['Data_quality']=utility.get_rg_data(I_sas.get_rg_for_plot())
-			Template_Dict['validation_input']=utility.get_rg_data_fits(I_sas.get_fits_for_plot())
-			if len(Template_Dict['validation_input'])<1:
-				Template_Dict['validation_input']=['Fit of model to data has not been deposited']
-			sas_data=I_sas.get_rg_for_plot()
-			sas_fit=I_sas.get_fits_for_plot()
-		else:
-			sas_data={}
-			sas_fit={}
-		return Template_Dict,sas_data,sas_fit
+    def run_model_quality(self, Template_Dict: dict) -> (dict, dict, dict, dict, dict):
+        '''
+        get excluded volume for multiscale models
+        get molprobity info for atomic models
+        exception: models with DNA--we need a way to assess models with DNA
+        '''
+        if self.Input.check_sphere() < 1:
+            # global clashscore; global rama; global sidechain;
+            exv_data = None
+            I_mp = molprobity.GetMolprobityInformation(self.mmcif_file)
+            filename = os.path.abspath(os.path.join(
+                os.getcwd(), '../static/results/', str(Template_Dict['ID'])+'_temp_mp.txt'))
+            global clashscore
+            global rama
+            global sidechain
 
-	def run_sas_validation_plots(self,Template_Dict:dict):
-		'''
-		get sas validation information from SASCIF or JSON files
-		'''
-		if self.I.check_for_sas(self.I.get_dataset_comp()):
-			Template_Dict['sas']=["True"]
-			I_sas=sas.sas_validation(self.mmcif_file)
-			try:
-				I_sas_plt=validation.sas_plots.sas_validation_plots(self.mmcif_file)
-				I_sas.modify_intensity()
-				I_sas.get_pofr_errors()	
-				I_sas_plt.plot_multiple()
-				I_sas_plt.plot_pf()
-				I_sas_plt.plot_Guinier()
-				if Template_Dict['number_of_fits']>0:
-					I_sas_plt.plot_fits()
-			except:
-				pass
+            if os.path.exists(filename):
+                d_mp = {}
+                print("Molprobity analysis file already exists...\n...assuming clashscores, \
+                        Ramachandran and rotamer outliers have already been calculated")
+                with open(filename, 'rb') as fp:
+                    d_mp['molprobity'] = pickle.load(fp)
+                f_rota = os.path.abspath(os.path.join(
+                    os.getcwd(), '../static/results/', str(Template_Dict['ID'])+'_temp_rota.txt'))
```

Comment:
You don't need the `os.getcwd()` here since that's essentially what `abspath` does.